src/bin.ts

```diff
@@ -50,8 +52,20 @@ export function main() {
 }
 
 function isTsEsmLoaderNeeded(cmd: string, args: string[]) {
-  // HACK: Ugly way to check if it's ts-node instead of node
-  return [cmd, ...args].includes("ts-node");
+  // HACK: Ugly way to check
+  const tsConfigTsNodeEsmIsTrue = () => {
+    const tsConfigPath = join(cwd(), "tsconfig.json");
```

Comment:
Should use `config.root` instead of `cwd()`. Also, it should search up — tsconfig could be in an ancestor directory.